packages/flutter_test/lib/src/controller.dart

```diff
@@ -31,12 +225,33 @@ const String _defaultPlatform = kIsWeb ? 'web' : 'android';
 ///
 /// Concrete subclasses must implement the [pump] method.
 abstract class WidgetController {
-  /// Creates a widget controller that uses the given binding.
-  WidgetController(this.binding);
+  /// Creates a widget controller that uses the given binding. A [SemanticsController]
+  /// will be created for [WidgetController.semantics] by default, but a custom
+  /// implementation can be provided for advanced use cases.
+  WidgetController(this.binding)
+    : _semantics = SemanticsController._(binding);
 
   /// A reference to the current instance of the binding.
   final WidgetsBinding binding;
 
+  /// Provides access to a [SemanticsController] for testing anything related to
+  /// the [Semantics] tree. Accessibility services, search engines, and other
```

Comment:
Our style guide suggests the first paragraph should be a brief description in a single sentence.

Reply:
s/Accessibility services/assistive technologies. That's what flutter uses to refer to talkback/voiceover.

Reply:
I pulled the "accessibility tools, search engines, and other analysis tools" part straight from the [`Semantics` docs](https://api.flutter.dev/flutter/widgets/Semantics-class.html). If that verbiage has changed to "assistive technologies" since then, should I update the `Semantics` docs as well?

I did at least update it from "accessibility services" to "accessibility tools" to match those docs, but if we want to change both, I'm good with that too.

Reply:
yes, I think `Semantics` doc pre-dates the decision when we decided to call all these things `assistive technologies`. We should update it if we find other places use a different name

Reply:
Perfect, thanks! I'll update that in my stuff and the `Semantics` class and ping here when that's done.

Reply:
Terminology updated, and consistent between `Semantics` and this work. Thanks for pointing this out!